bbq.py

```diff
@@ -568,41 +403,213 @@ def calc_Pjs_from_I_for_mode(self,variation, U_H,U_E, LJs, junc_rects,junc_lens,
+                self.solutions.set_mode(mode+1, 0)
+                self.fields = self.setup.get_fields()
+
+                print_NoNewLine('   U_H ...');     sol['U_H'] = self.U_H = self.calc_U_H(variation)
+                print_NoNewLine('   U_E');         sol['U_E'] = self.U_E = self.calc_U_E(variation)
+                print(  "   =>   U_L = %.3f%%" %( (self.U_E - self.U_H )/(2*self.U_E)) )
+                
+                if self.Pj_from_current:
+                    self.LJs    = [ ureg.Quantity(varz['_'+LJvar_nm]).to_base_units().magnitude  for LJvar_nm in junc_LJ_var_name]
+                    meta_data['LJs'] = dict(zip(junc_LJ_var_name, self.LJs))
+                    print '   I -> p_{mJ} ...'
+                    sol_PJ = self.calc_Pjs_from_I_for_mode(variation, self.U_H, self.U_E, self.LJs, junc_rect, junc_len, 
+                                                                 method = pJ_method, freq = freqs_bare_vals[mode]*10**-9,
+                                                                 calc_sign = junc_lines)
+                    sol = sol.append(sol_PJ)
+                
+                if self.njunc == 1:             # Single-junction method using global U_H and U_E; 
+                    assert(type(junc_LJ_var_name) == list and len(junc_LJ_var_name) == 1), "Please pass junc_LJ_var_name as array of 1 element for a single junction; e.g., junc_LJ_var_name = ['junc1']" 
+                    #lj  = 1E-3*ureg.Quantity(varz['_'+junc_LJ_var_name]).to_base_units().magnitude                        
+                    sol['pj1'] = self.get_p_j(mode)
+                    self.pjs.update(sol['pj1'])        # convinience function for single junction case
+                    
+                if seams is not None:           # get seam Q
+                    for seam in seams: sol = sol.append(self.get_Qseam(seam,mode,variation))
+
+                if dielectrics is not None:     # get Q dielectric      
+                    for dielectric in dielectrics: sol = sol.append(self.get_Qdielectric(dielectric, mode, variation))
+                               
+                if surface is True:             # get Q surface                              
+                    sol = sol.append( self.get_Qsurface(mode, variation) )
+                    
+                var_sol_accum +=[sol]
+            
+            #TODO: add metadata to the Dataframe & save it
+            #      such as what are the junc_rect names and Lj values etc.  (e.g., http://stackoverflow.com/questions/29129095/save-additional-attributes-in-pandas-dataframe/29130146#29130146)
+            hdf[variation+'/eBBQ_solution']  = self.sols[variation]  \
+                                             = pd.DataFrame(var_sol_accum, index = modes)            
+            hdf[variation+'/meta_data']      = self.meta_data[variation]  \
+                                             = Series(meta_data)
```

Comment:
basically there is 3 things now that get returned 
hfss_variables
eBBQ_solution
meta_data